File: src/modules/user-management/domain/events/driver-availability-changed.event.ts
import { BaseDomainEvent } from '@shared/domain/events/base-domain-event';

export class DriverAvailabilityChangedEvent extends BaseDomainEvent {
  public readonly eventName = 'driver.availability-changed';
  constructor(
    public readonly driverProfileId: string,
    public readonly userId: string,
    public readonly isAvailable: boolean,
  ) { super(); }
}

File: src/main/java/com/dddrivebye/usermanagement/domain/event/DriverAvailabilityChangedEvent.java
package com.dddrivebye.usermanagement.domain.event;

import com.dddrivebye.shared.domain.event.BaseDomainEvent;
import com.dddrivebye.usermanagement.domain.valueobject.AvailabilityStatus;
import com.dddrivebye.usermanagement.domain.valueobject.UserId;

public final class DriverAvailabilityChangedEvent extends BaseDomainEvent {

    private final UserId userId;
    private final AvailabilityStatus previousStatus;
    private final AvailabilityStatus newStatus;

    public DriverAvailabilityChangedEvent(UserId userId, AvailabilityStatus previousStatus, AvailabilityStatus newStatus) {
        this.userId = userId;
        this.previousStatus = previousStatus;
        this.newStatus = newStatus;
    }

    public UserId userId() {
        return userId;
    }

    public AvailabilityStatus previousStatus() {
        return previousStatus;
    }

    public AvailabilityStatus newStatus() {
        return newStatus;
    }
}
